dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -103,6 +107,140 @@ TEST_F(FramesDecoderTest, VariableFrameRate) {
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 0), file.FrameSize());
 }
 
+TEST_F(FramesDecoderTest, VariableFrameRateGpu) {
+    ASSERT_TRUE(cuInitChecked());
+    CUdevice device = { 0 };
+    CUDA_CALL(cuDeviceGet(&device, 0));
+    CUcontext context = { 0 };
+    CUDA_CALL(cuCtxCreate(&context, 0, device));
```

Comment:
I assume you want to test on the first device not the one that the context was created for before calling that test?
```suggestion
    DeviceGuard(0);
```

Reply:
For this test context was not created before running it, so I did this manually. 

Maybe we can make some small utility in a separate PR and refactor it?

Reply:
`DeviceGuard` does that. It saves the current context (to improve the interoperability with libraries using driver API instead of CUDA API), and calls `cudaSetDevice` which creates the context. So unless you need to access this context directly `DeviceGuard` should do.
I suspect that `cuDeviceGet` may also create the context, but I'm not sure.

Reply:
This is a rare corner case where `DeviceGuard` on its own does not work, unfortunately.
That is because `cudaSetDevice` does not create the context. It sets primary device context as the current context. Primary context is created upon first call to any CUDA Runtime function. 
Here, the first call we make is to NVDECODE API and it fails due to lack of context.
That is why I needed to create it manually. 

I changed it to:
```
DeviceGuard(0);
CUDA_CALL(cudaDeviceSynchronize());
```
This way `cudaDeviceSynchronize` will create the context and the whole thing works.